lib/index.d.ts

```diff
@@ -0,0 +1,182 @@
+declare module 'node-torrent' {
+  import { Stream } from 'stream';
+
+  /**
+   * A Hasher object is returned when a torrent is created with `make` and when `Torrent#hashCheck` is called. It inherits from ReadableStream.
+   */
+  class Hasher extends Stream {
+    /**
+     * Pause hash checking.
+     */
+    pause(): boolean;
+
+    /**
+     * Resumes hash checking.
+     */
+    resume(): boolean;
+
+    /**
+     * Continues hashing if paused or pauses if not.
+     */
+    toggle(): boolean;
+
+    /**
+     * Stops hashing completely. Closes file descriptors and does not emit any more events.
+     */
+    destroy(): boolean;
+  }
+
+  /**
+   * The `read` and `make` functions all call their callback with a Torrent object.
+   */
+  class Torrent {
+    /**
+     * Contains metadata of the torrent. Example:
+     *
+     * ```js
+     * {
+     *   announce: 'udp://tracker.publicbt.com:80',
+     *   'announce-list': [
+     *     [ 'udp://tracker.publicbt.com:80' ],
+     *     [ 'udp://tracker.ccc.de:80' ],
+     *     [ 'udp://tracker.openbittorrent.com:80' ],
+     *     [ 'http://tracker.thepiratebay.org/announce' ]
+     *   ],
+     *   comment: 'Torrent downloaded from http://thepiratebay.org',
+     *   'creation date': 1303979726,
+     *   info: { length: 718583808,
+     *     name: 'ubuntu-11.04-desktop-i386.iso',
+     *     'piece length': 524288,
+     *     pieces: <Buffer e5 7a ...>
+     *   }
+     * }
+     * ```
+     */
+    metadata: object;
```

Comment:
do you mind adding the details of the metadata object? looking at `schema.js` it looks something like

```js
{
  announce: string;
  'announce-list'?: string[][]
  'creation-date'?: string;
  comment?: string;
  'created-by'?: string;
  encoding?: string;
  info: MultiFileInfo | SingleFileInfo;
};

type Info = {
    name?: string;
    'piece-length': int;
    pieces: string;
    private?: 0 | 1;
    source?: string;
  };
type MultiFileInfo = Info & {
  files: {
    length: int;
    path: string;
    md5sum?: string;
  }[];
}
type SingleFileInfo = Info & {
  // maybe it needs `name` too?
  length: int;
  md5sum?: string;
}
```

Reply:
Thank you for typing it out for me. :+1: I think if `SingleFileInfo` includes the `Info` type then it won't need `name` specified